app/uk/gov/hmrc/transitmovements/services/DeparturesFactory.scala

```diff
@@ -44,68 +46,63 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.util.control.NonFatal
 
-@ImplementedBy(classOf[DeparturesServiceImpl])
-trait DeparturesService {
+@ImplementedBy(classOf[DeparturesFactoryImpl])
+trait DeparturesFactory {
 
   def create(
     eori: EORINumber,
     declarationData: DeclarationData,
     tempFile: Source[ByteString, Future[IOResult]]
-  ): EitherT[Future, MongoError, DeclarationResponse]
+  ): EitherT[Future, StreamError, Departure]
 }
 
-class DeparturesServiceImpl @Inject() (
-  repository: DeparturesRepository,
+class DeparturesFactoryImpl @Inject() (
   clock: Clock,
   random: SecureRandom
 )(implicit
   val materializer: Materializer
-) extends DeparturesService {
+) extends DeparturesFactory {
 
   def create(
     eori: EORINumber,
     declarationData: DeclarationData,
     tempFile: Source[ByteString, Future[IOResult]]
-  ): EitherT[Future, MongoError, DeclarationResponse] =
-    for {
-      messageBody <- getMessageBody(tempFile)
-      departure = createDeparture(eori, declarationData, Some(messageBody))
-      _ <- repository.insert(departure)
-    } yield DeclarationResponse(departure._id, departure.messages.head.id)
-
-  private def createDeparture(eori: EORINumber, declarationData: DeclarationData, messageBody: Option[String]): Departure =
-    Departure(
-      _id = DepartureId(ShortUUID.next(clock, random)),
-      enrollmentEORINumber = eori,
-      movementEORINumber = declarationData.movementEoriNumber,
-      movementReferenceNumber = None,
-      created = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC),
-      updated = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC),
-      messages = Seq(createDeclarationMessage(declarationData, messageBody))
-    )
+  ): EitherT[Future, StreamError, Departure] =
+    getMessageBody(tempFile).map {
+      message =>
+        Departure(
+          _id = DepartureId(ShortUUID.next(clock, random)),
+          enrollmentEORINumber = eori,
+          movementEORINumber = declarationData.movementEoriNumber,
+          movementReferenceNumber = None,
+          created = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC),
+          updated = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC),
+          messages = NonEmptyList.one(createDeclarationMessage(declarationData, message))
+        )
+    }
 
-  private def createDeclarationMessage(declarationData: DeclarationData, messageBody: Option[String]): MovementMessage =
-    MovementMessage(
-      id = MovementMessageId(ShortUUID.next(clock, random)),
+  //To Consider: Refactor to a MessageFactory?
```

Comment:
We're expecting the message object to be the same across departures and arrivals, so yes, I think that would be wise - but this would need `generationDate` and `messageType` to become parameters.

Is it reasonable to do it now while fresh in heads, or is this a note we need to carry forward?

Reply:
I think it's best to take it forward to when we persist other message types. It'll need to be usable in both this scenario and when we create other messages, and it'll be clearer at that time.